### ISSUE TITLE
Parsing Error with [ in !Sub

### ISSUE THREAD

**Issue by nappa32** [realestate-com-au/stackup#39]
I'm trying to create a CFN init object with Files like this : 

```
          files:
            /etc/awslogs/awscli.conf:
              content: !Sub: |
                  [default]
                  region = ${AWS::Region}
                  [plugins]
                  cwlogs = cwlogs

```
This will break the parsing, and cause the gem to spit out the entire thing. 

```
  /home/imran/.rbenv/versions/2.3.1-railsexpress/lib/ruby/2.3.0/json/common.rb:156:in `parse'
  /home/imran/.rbenv/versions/2.3.1-railsexpress/lib/ruby/2.3.0/json/common.rb:156:in `parse'
  /home/imran/.rbenv/versions/2.3.1-railsexpress/lib/ruby/gems/2.3.0/gems/multi_json-1.12.1/lib/multi_json/adapters/json_common.rb:14:in `load'
  /home/imran/.rbenv/versions/2.3.1-railsexpress/lib/ruby/gems/2.3.0/gems/multi_json-1.12.1/lib/multi_json/adapter.rb:21:in `load'
  /home/imran/.rbenv/versions/2.3.1-railsexpress/lib/ruby/gems/2.3.0/gems/multi_json-1.12.1/lib/multi_json.rb:122:in `load'


```

But if I put a \ infront of the [ - seems to work.

**Comment by mdub** [comment]
From the backtrack, it appears to be attempting to parse your template as JSON, rather than YAML, which is pretty strange.

**Comment by mdub** [comment]
I'm pretty sure this is just a dodgy regex. In [lib/stackup/source.rb:32](https://github.com/realestate-com-au/stackup/blob/4bb7350c0fb44ae240ebb3d936939bd78bd7abb6/lib/stackup/source.rb#L32) we find:

    LOOKS_LIKE_JSON = /^\s*[\{\[]/

Rather than anchoring on beginning of line (with `^`), we should be anchoring on beginning-on-string (with `\A`).

**Comment by mdub** [comment]
Fix released in version 1.1.3.
